### PR TITLE
Holiday off overtime

### PR DESCRIPTION
**Pull request by AdyT14**
Completed implementing the features for adding overtime and free day for a specific vendor
https://xabriel.atlassian.net/browse/DBVEN-29
https://trello.com/c/XS9VtHsG/54-admin-module-add-required-methods-for-overtime-days-off-be

### CODE REVIEW

File: src/test/java/ro/db/vendor/service/HolidayServiceImplTest.java [hunk 1]
```diff
@@ -25,32 +28,57 @@
...
+    holiday.setStartDate(offDayModel.getStartDate());
+    holiday.setEndDate(offDayModel.getEndDate());
+    BDDMockito.when(holidayRepositoryMock.save(holiday)).thenReturn(holiday);
 
-    Holidays holidays = holidayService.insertHoliday(holidayToInsert);
+    boolean insertHoliday = holidayService.insertHoliday(offDayModel);
 
-    assertEquals(123, holidays.getVendorsByIdVendor().getIdVendor());
-    assertEquals(timeStamp, holidays.getStartDate());
+    assertEquals(true, true);
```
Comment (alexandrubirlan) [inline src/test/java/ro/db/vendor/service/HolidayServiceImplTest.java]: :) you already know what's wrong here, don't you?

File: src/test/java/ro/db/vendor/service/HolidayServiceImplTest.java [hunk 2]
```diff
@@ -25,32 +28,57 @@
...
   @Before
   public void setUp() throws Exception {
     MockitoAnnotations.initMocks(this);
-    holidayService = new HolidayServiceImpl(holidayRepositoryMock);
+    holidayService = new HolidayServiceImpl(holidayRepositoryMock, vendorRepositoryMock,
+        vendorDaysOffRepositoryMock);
   }
 
   @Test
   public void insertHoliday() {
```
Comment (alexandrubirlan) [inline src/test/java/ro/db/vendor/service/HolidayServiceImplTest.java]: Also create test for negative scenario, i.e. when you cannot add a holiday because there are no more tests left.
Should assert on exception (use Expected = ... )

File: src/test/java/ro/db/vendor/service/HolidayServiceImplTest.java (hunk 1 above)
Comment (AdyT14) [reply src/test/java/ro/db/vendor/service/HolidayServiceImplTest.java]: Wow, i did not notice, sorry, fixing it now.

**Comment by AdyT14** [comment]
I made the required modifications, i think. Also realized that i made mistakes in implementing the method for insertHoliday. It was returning false when inserting a type other than "day_off".
